### PR TITLE
Trace view tool

### PR DESCRIPTION
**Pull request by snehasish**
Add a trace view tool which emits disassembly for an offline trace. The
format can be set to one of att (default), intel, dr or arm. Command
line flags -skip_refs and -sim_refs can be used to fast forward and only
emit a certain number of instructions from the trace respectively. This
also adds a basic e2e test.

### CODE REVIEW

File: clients/drcachesim/tools/view_create.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+#include "analysis_tool.h"
+
+/**
+ * @file drmemtrace/view.h
+ * @brief DrMemtrace view trace analysis tool creation.
+ */
+
+/**
+ * Creates an analysis tool which prints out the dissasembled instructions from
```
Comment (derekbruening) [inline clients/drcachesim/tools/view_create.h]: spelling: disassembled

File: suite/tests/CMakeLists.txt [hunk 2]
```diff
@@ -2647,6 +2647,10 @@ if (CLIENT_INTERFACE)
       set(tool.drcacheoff.opcode_mix_depends tool.drcacheoff.simple)
       set(tool.drcacheoff.opcode_mix_depends tool.drcacheoff.filter)
 
+      torunonly_drcacheoff(view ${ci_shared_app} ""
+        "@-simulator_type@view" "")
+      set(tool.drcacheoff.view_depends tool.drcacheoff.opcode_mix)
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: Please include a comment explaining why it depends.

File: clients/drcachesim/tools/view.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      // MAX_INSTR_DIS_SZ is set to 196 in core/arch/disassemble.h but is not
+      // exported so we just use the same value here.
+      char buf[196];
+      instr_disassemble_to_buffer(dcontext, &instr, buf, 196);
+      disasm = buf;
+      disasm_cache.insert({mapped_pc, disasm});
+      instr_free(dcontext, &instr);
+  }
+  std::cerr << disasm << "\n";
+  ++num_disasm_instrs;
```
Comment (derekbruening) [inline clients/drcachesim/tools/view.cpp]: Do you plan to also print out data load and store information in a future CL?  IMHO that would be useful.  Perhaps a `XXX` comment about it.

File: clients/drcachesim/tools/view.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    virtual bool print_results();
+
+ protected:
+    void *dcontext;
+    raw2trace_t *raw2trace;
+    raw2trace_directory_t directory;
+    unsigned int knob_verbose;
+    uint64_t instr_count;
+    static const std::string TOOL_NAME;
+    uint64_t skip_refs;
```
[the remark this answers is not in the document]
Comment (snehasish) [reply clients/drcachesim/tools/view.h]: Done.

File: clients/drcachesim/tools/view.h [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    uint64_t sim_refs;
```
[the remark this answers is not in the document]
Comment (snehasish) [reply clients/drcachesim/tools/view.h]: Done.

File: clients/drcachesim/tools/view.cpp [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+      instr_t instr;
+      instr_init(dcontext, &instr);
+      app_pc next_pc = decode(dcontext, mapped_pc, &instr);
+      if (next_pc == NULL || !instr_valid(&instr))
+          return false;
+
+      // MAX_INSTR_DIS_SZ is set to 196 in core/arch/disassemble.h but is not
+      // exported so we just use the same value here.
+      char buf[196];
+      instr_disassemble_to_buffer(dcontext, &instr, buf, 196);
```
[the remark this answers is not in the document]
Comment (snehasish) [reply clients/drcachesim/tools/view.cpp]: Done.

File: clients/drcachesim/tools/view.cpp [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  if (!err.empty())
+      return false;
+
+  std::string disasm;
+  auto cached_disasm = disasm_cache.find(mapped_pc);
+  if (cached_disasm != disasm_cache.end()) {
+      disasm = cached_disasm->second;
+  } else {
+      instr_t instr;
+      instr_init(dcontext, &instr);
```
[the remark this answers is not in the document]
Comment (snehasish) [reply clients/drcachesim/tools/view.cpp]: Done.

File: clients/drcachesim/tools/view_create.h (hunk 1 above)
Comment (snehasish) [reply clients/drcachesim/tools/view_create.h]: Done.

File: clients/drcachesim/tools/view.cpp (hunk 3 above)
Comment (snehasish) [reply clients/drcachesim/tools/view.cpp]: Done.

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (snehasish) [reply suite/tests/CMakeLists.txt]: Done.

**Pull request merged**